jupyterhub/app.py

```diff
@@ -864,12 +865,13 @@ def cleanup(self):
         
         # clean up proxy while SUS are shutting down
         if self.cleanup_proxy:
-            if self.proxy_process and self.proxy_process.poll() is None:
+            if self.proxy_process:
                 self.log.info("Cleaning up proxy[%i]...", self.proxy_process.pid)
-                try:
-                    self.proxy_process.terminate()
-                except Exception as e:
-                    self.log.error("Failed to terminate proxy process: %s", e)
+                if self.proxy_process.poll() is None:
+                    try:
+                        self.proxy_process.terminate()
+                    except Exception as e:
+                        self.log.error("Failed to terminate proxy process: %s", e)
             else:
                 self.log.info("I didn't start the proxy, I can't clean it up")
```

Comment:
the changes here are because `I didn't start the proxy` log message was firing because SIGTERM shutdown the proxy as well, and proxy_process.poll() was returning the exit code.